lib/services/metadata_helper.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:device_info_plus/device_info_plus.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:get_it/get_it.dart';
+import 'dart:io';
+
+final getIt = GetIt.instance;
+
+class DeviceInfo {
+  late String osVersion;
+  late String deviceModel;
+  late String deviceType;
+
+  DeviceInfo();
+
+  Future<void> init() async {
+    DeviceInfoPlugin deviceInfo = DeviceInfoPlugin();
+    if (Platform.isAndroid) {
+      AndroidDeviceInfo androidInfo = await deviceInfo.androidInfo;
+      osVersion = androidInfo.version.release;
+      deviceModel = androidInfo.model;
+      deviceType =
+          "Phone"; // You can add more logic to determine the device type
+    } else if (Platform.isIOS) {
+      IosDeviceInfo iosInfo = await deviceInfo.iosInfo;
+      osVersion = iosInfo.systemVersion;
+      deviceModel = iosInfo.name;
+      deviceType =
+          "Phone"; // You can add more logic to determine the device type
+    } else if (Platform.isLinux) {
+      LinuxDeviceInfo linuxInfo = await deviceInfo.linuxInfo;
+      osVersion = linuxInfo.version ?? 'Unknown';
+      deviceModel = linuxInfo.id;
+      deviceType = "PC"; // You can add more logic to determine the device type
+    } else if (Platform.isWindows) {
+      WindowsDeviceInfo windowsInfo = await deviceInfo.windowsInfo;
+      osVersion = windowsInfo.displayVersion;
+      deviceModel = windowsInfo.deviceId;
+      deviceType = "PC"; // You can add more logic to determine the device type
```

Comment:
how about you just add `platform` instead of device type, and set them accordingly (Android, Windows, Linux, etc.)?